### DISCUSSION TITLE
Bounty Hunter Protection

### DISCUSSION THREAD

**Discussion by 0x4007** [ubiquity/ubiquibot#279] [Ideas]
I think it could be interesting to create some type of incentive system to protect bounty hunters from teams that take too long to finish reviews and approve bounties

---

              > Do you have any updates? @zgorizzo69

Waiting for fam to review pr #600

_Originally posted by @zgorizzo69 in https://github.com/ubiquity/ubiquity-dollar/issues/593#issuecomment-1467558055_

**Comment by sergfeldman** [comment]
Bounty hunter protection is an interesting idea.
I have been thinking about similar ideas.
Let's discuss this.

1. **Protection fund** 
- Protects bounty hunters from projects that are not effective in terms of bounty review (need to track metrics like time of PR review, etc)

- Protects projects from bounty hunters that are not effective in terms of bounty implementation (need to track metrics that we discussed for the well done bounty - according to time, not high number of PRs, etc)

- If a bounty hunter or a project are not effective to not meet (violate) some thresholds of metrics then they should pay a fee in UBQ (or uAD as a part of their bounty) into this Protection fund.

- Fee in UBQ can be in fixed amount, for example 100 UBQ for 1 bounty. Fee in uAD can be a percentage of the bounty price, for example 10%. In the future, additional fees can be applied to parties that violated were not defective in several bounties.

- If a bounty hunter or a project is suffered from a not effective partner then he should be able to get compensation in UBQ from this Protection fund.


2. **Lottery** 

If a bounty hunter is suffered from a project who slowly reviews his PR, then each day he can win a bonus in UBQ.

A bonus in UBQ is randomly varies from 1 to 1000 UBQ. 
The longer he waits the higher chances to win a large bonus.


3. **Bounty efficiency standard and report**

Ubiquity can define a kind of a golden standard for the bounties.
The standard will consist of the best practices and metrics to track.
We already discussed the some examples of metrics - time, number of PRs, etc

Then Ubiquity can scrape data on the all (DevPool and others bounties from the GitHub and publish regular Bounty state report (currently, not clear how to recognise the bounty).

Bounty state report will show current bounty resolution metrics and trends:
- Number of projects who offer bounties
- Number of open and done bounties
- Average bounty price
- And other metrics

Such reports are published by the companies who want to position as leaders on this market
- Google - DORA and other reports
- Alchemy - Web3 development [https://www.alchemy.com/blog/web3-developer-report-q4-2022](https://www.alchemy.com/blog/web3-developer-report-q4-2022)

The further development of the Bounty standard and support of the Bounty state report can be a way to apply for non-profit grants.
